src/libvirtApi/domain.ts

```diff
@@ -113,6 +113,234 @@ function buildConsoleVVFile(consoleDetail: VMGraphics): string {
         'fullscreen=0\n';
 }
 
+function spawn(connectionName: ConnectionName, args: string[]): cockpit.Spawn<string> {
+    return cockpit.spawn(
+        args,
+        {
+            err: "message",
+            ...(connectionName === "system" ? { superuser: "try" } : { })
+        });
+}
+
+function script(connectionName: ConnectionName, script: string): cockpit.Spawn<string> {
+    return cockpit.script(
+        script,
+        undefined,
+        {
+            err: "message",
+            ...(connectionName === "system" ? { superuser: "try" } : { })
+        });
+}
+
+/* Running virt-xml
+
+   The virtXmlAdd, virtXmlEdit, etc family of functionscan be used to
+   run virt-xml with various actions, options, and values. A typical
+   invocation looks like this:
+
+       virtXmlAdd(vm, "graphics", { type: "vnc", port: 5900 });
+
+   This will result in running
+
+       virt-xml <vm-uuid> --add-device --graphics type=vnc,port=5900
+
+   The values are turned into the CSV style of virt-xml in the
+   'obvious' way, according to these rules:
+
+   Properties with "null", "false", or "undefined" as their values are
+   not added to the result at all.  The value "true" is added as the
+   string "yes".
+
+       { type: null, password: "", tls: true }
+       ->
+       password=,tls=yes
+
+   A property name can be the empty string. In that case its value is
+   added with the parent name.
+
+       { source: { "": "abc", mode: "123" } }
+       ->
+       source=abc,source.mode=123
+
+   Arrays are encoded by appending consecutive numbers to the parent.
+
+       { port: [ "a", "b", "c" ] }
+       ->
+       port0=a,port1=b,port2=c
```

Comment:
These two are some seriously weird magic that I'd rather keep in the caller to make it more obvious. WDYT?